tests/test_addons_toupdate.py

```diff
@@ -0,0 +1,610 @@
+
+
+def test_addon_toupdate_18(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - remove odoo addon dependency
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, 'bfde22f2c292ecc2921e1b7fc09b31bdf79b1f82')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '3382b4e8afb086b0aaa4946f9bf9265c7a41c768',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_19(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - remove library dependency
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '657f0f91012b35772c28e1d3ad936caada3ed64c')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            'bfde22f2c292ecc2921e1b7fc09b31bdf79b1f82',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_20(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - change editable odoo addon dependency URL
+        - compare requirements: YES
+    Expected result:
+        - modified odoo addon's name (base_user_role)
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '77d64b380dda00824f525d65f22d0677eb63eead')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '72d6e17f5ae7df754c3bdad6d4603a9c0bc382c0',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = 'base_user_role\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_21(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - requirements file exists
+    Test case:
+        - compared ref is not an ancestor of current ref
+        - compare requirements: NO
+    Expected result:
+        - exit code != 0
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, 'cfdb4d310cd04ce0ea8098586007f7d561a1c0aa')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '5ce68ab261be5c280318f4c1c21545c6196c8438',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 != res.exit_code
```

Comment:
By reading the code above, I understood it would return `all` and suceed in this case.